src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -90,17 +91,21 @@ public async Task InvokeAsync(HttpContext context)
                 switch (mediaTypeHeader.MediaType)
                 {
                     case MediaType.JSON:
-                        var deserializationResult = await _deserializer.DeserializeFromJsonBodyAsync(httpRequest, cancellationToken);
-                        if (!deserializationResult.IsSuccessful)
+                        GraphQLRequest[] deserializationResult;
+                        try
                         {
-                            var message = deserializationResult.Exception is null
-                                ? "JSON body text could not be parsed."
-                                : $"JSON body text could not be parsed. {deserializationResult.Exception.Message}";
+                            deserializationResult = await _serializer.ReadAsync<GraphQLRequest[]>(httpRequest.Body, cancellationToken);
+                        }
+                        catch (Exception ex)
```

Comment:
Exception is swallowed here. Let's add `protected virtual bool HandleDeserializationError(Exception e) => true`
And here (before return):
```csharp
if (!HandleDeserializationError(ex))
    throw;
```

Reply:
Also `GraphQLHttpMiddlewareWithLogs` is a good example to override `HandleDeserializationError` to write log.

Reply:
I don't mind but I don't understand how that makes sense.  If it fails to parse the JSON, execution cannot continue.  There is no query to execute.  So it is not possible to "handle" a JSON exception.

I'll add a void protected method instead.

Reply:
Oh wait so handle true makes it return a failure instead of returning an error?  I'm still just as confused - how does it make sense to rethrow the exception?

Still going to add a void method.

Reply:
1) `HandleDeserializationError` as it is can log exception, for example.
2) Next, if `HandleDeserializationError` returns `bool` then one can decide whether to throw and return some other HTTP related result or just return _standard_ result. Imagine `UseDeveloperExceptionPage` somewhere in the request pipeline. Now any deserialization exception is swallowed, we return 400 code, the end. I want to add the possibility to bubble occurred exception up.

Reply:
#719

Reply:
ok